app/models/concerns/letters_and_calls_costs.rb

```diff
@@ -74,10 +74,10 @@ def letters_and_calls_total_cost_inc_vat
   private
 
   def letters_before_uplift
-    letters.to_d * pricing.letters if letters && !letters.zero?
+    letters.to_d * pricing.letters if letters && !letters.to_i.zero?
   end
 
   def calls_before_uplift
-    calls.to_d * pricing.letters if calls && !calls.zero?
+    calls.to_d * pricing.letters if calls && !calls.to_i.zero?
```

Comment:
I added the to_i on the 2nd part of the clauses because with the changes to the form (use can put in non numeric characters into the field), these methods will cause a 500 error if you try to run a calculation via clicking the calculate changes button (as this is pre-validation). So adding the to_i makes it so that invalid entries just change to 0 instead of error-ing out 

My assumption re: the .to_d (which was already there), is so that the type of the calls value matches the type of the pricing value (which is a decimal). A decimal multiplied by an integer _should_ ouput a decimal - but I guess this just makes it more secure if Ruby decides to be weird and change how that works?

Reply:
Re: the first paragraph - there could potentially be a better experience re: the calculate changes but thought that should be a separate ticket